src/System.ServiceProcess.ServiceController/src/System/ServiceProcess/ServiceBase.cs

```diff
@@ -889,10 +915,32 @@ public unsafe void ServiceMainCallback(int argCount, IntPtr argPointer)
 
         private void WriteLogEntry(string message, bool error = false)
         {
-            // Used to write to EventLog but for now just logging to debug output stream
-            // might want to plumb other logging in the future.
-
-            Debug.WriteLine((error ? "Error: " : "") + message);
+            // EventLog failures shouldn't affect the service operation
+            try 
+            {
+                if (AutoLog)
+                {
+                    EventLog.WriteEntry(message); 
+                }
+            }
+            #region Stuff not to catch
+            catch (StackOverflowException) 
```

Comment:
You can remove SOE .. it's a "corrupting state exception" and CoreCLR won't let you catch SOE unless you threw them yourself.

Reply:
AccessViolatoinException is in the same category. I am not sure what else. [This](https://msdn.microsoft.com/en-us/magazine/dd419661.aspx) suggests it is all SEH exceptions but I am not sure what that means, DivideByZeroException is presumably catchale. @jkotas is there a list?

Reply:
I think you can remove everything here, and just do `catch { }`